### DISCUSSION TITLE
Navigation Query Builder and in Blade

### DISCUSSION THREAD

**Discussion by Dimchao** [statamic/cms#8498] [Troubleshooting]
Does anyone know how to run a query builder command for the Statamic::tag('nav:my_navigation)? I can do it for collections:pages, but for nav, it doesn't do anything.

So far:
`Statamic::tag('collection:pages')->where('statis:is',` 'published')->fetch();`
 works as expected.

However:
`Statamic::tag('nav:my_navigation')->where('param:is', 'my_value')->fetch();`
 does nothing but return everything in the 'nav:my_navigation'.

Is this not a feature? The help docs mention that you can do this in antlers, but is full on not working/not implemented otherwise?

New to Statamic, and this seems like user error on my end...?

Basically, I will be handing off the CMS to the client, and it makes sense for all of their menus to be in the 'Navigation' sidebar. However, when creating a toggle field 'enabled', looping through it is a pain considering I have to use $menuItem->enabled->value(), which throws an error if enabled isn't set in the yaml files... which is something that happens when saving a toggle field that is set to false.

**Comment by jasonvarga** [comment]
What actual parameter are you trying to use?

When using the blade syntax, the method needs to either be the parameter, or a `param()`. You can't just use `where`.

```php
Statamic::tag('collection:pages')->limit(3);
// or
Statamic::tag('collection:pages')->param('limit', '3');
```

In your example you should be doing:

```php
->param('param:is', 'my_value')
```


Your first example doesn't work either. It just seems to work because only published items are shown by default.

**Reply by Dimchao** [reply]
@jasonvarga I was hoping to find a way to use a query builder for items in the Navigation. I can pull all of the navigation items using: `Statamic::tag('nav:main_menu')->fetch();`
However, this doesn't seem to do anything when I try and use the 'param' or 'where' chained methods.

**Reply by jasonvarga** [reply]
What *actual* parameter are you trying to use? The real one, not just an example.

**Reply by Dimchao** [reply]
@jasonvarga

`$menuItems = Statamic::tag('nav:main_menu')->params(['enabled:is' => true])->fetch();`

'enabled' was set as a toggle field in the blueprint for the Nav items in the Navigation section of the Statamic CMS.

**Reply by jasonvarga** [reply]
Is this an actual "navigation" or is it a collection?

Just looking into it more, you can only perform filtering when it's a collection. Sorry, forgot about that.

It's not supported on navs since the intention is that you already hand-crafted the exact navigation tree you needed.

**Reply by Dimchao** [reply]
@jasonvarga Ah. I just know that I am handing this project off to a client who does not know how to program. The idea is that they can build out navigation trees as they need them, and then enable/disable any navigation items they need, or even schedule them to be turned on. This is for a performing arts company which has different needs. It would be great to have the navigation follow the same convention as any other collection when using Statamic::tag.
